### ISSUE TITLE
The "-j" option does not report any error message by zsync client

### ISSUE THREAD

**Issue by pamarcos** [AppImage/AppImageUpdate#56]
Since for the "-j" option the zsync client is created on the stack (https://github.com/AppImage/AppImageUpdate/blob/rewrite/src/updater.cpp#L520) and not assigned to `zsyncClient`, there's no way to currently pull for error messages https://github.com/AppImage/AppImageUpdate/blob/rewrite/src/updater.cpp#L618.

**Comment by TheAssassin** [comment]
Two options to solve this:

a) rethink the role of the `zSyncClient`, we most likely don't need an extra instance to avoid modifications of its state
b) push the messages to the updater's message queue after calling `checkForChanges` (would be some kind of duplicate code, though)

**Comment by TheAssassin** [comment]
There was a lot of fiddling required for resolving this issue. I added some additional complexity, but I tried to keep the amount of changes rather minimal. It seems to work fine.

I also changed the default behavior to always display the text even if the update check works (whether there's an update or not doesn't matter here). I thought like, if you don't want/need the information, just don't use it.
